app/src/main/java/org/mozilla/focus/screenshot/CaptureRunnable.java

```diff
@@ -98,7 +93,13 @@ private void promptScreenshotResult(int snackbarTitleId) {
             return;
         }
         Toast.makeText(context, snackbarTitleId, Toast.LENGTH_SHORT).show();
-        completed = true;
+        if (refBrowserFragment != null &&
```

Comment:
It looks `refBrowserFragment.get() != null` check is duplicated. How about simplifying to 

if (refBrowserFragment != null) {
            final BrowserFragment browserFragment = refBrowserFragment.get();
            if (browserFragment != null && browserFragment.getCaptureStateListener() != null) {
                browserFragment.getCaptureStateListener().onPromptScreenshotResult();
            }
       }

Reply:
fixed